lib/glific/templates/interactive_templates.ex

```diff
@@ -676,44 +676,39 @@ defmodule Glific.Templates.InteractiveTemplates do
 
   @spec do_items_translated(map(), list()) :: list()
   defp do_items_translated(content, items_translations) do
-    chunk_sizes =
-      Enum.map(content["items"], fn item ->
+    {items, _remaining_translations} =
+      Enum.reduce(content["items"], {[], items_translations}, fn item,
+                                                                 {acc, remaining_translations} ->
         # option length is the number of elements in the options list inside items
-        # We multiply this by 2 because we are translating only the title and description,
+        # multiply this by 2 because we are translating only the title and description,
         # another 2 that we are adding is for the subtitle and title.
-        2 + length(item["options"]) * 2
-      end)
-
-    translations_chunks =
-      Enum.reduce(chunk_sizes, {[], items_translations}, fn chunk_size,
-                                                            {acc, remaining_translations} ->
+        chunk_size = 2 + length(item["options"]) * 2
         {chunk, rest} = Enum.split(remaining_translations, chunk_size)
-        {acc ++ [chunk], rest}
+
+        [item_title, item_subtitle | options_translations] = chunk
+
+        options =
+          options_translations
+          |> Enum.chunk_every(2)
+          |> Enum.zip(item["options"])
+          |> Enum.map(fn {[option_title, option_description], option} ->
+            %{
+              "title" => option_title,
+              "description" => option_description,
+              "type" => option["type"]
+            }
+          end)
```

Comment:
We can do a similar Enum.reduce operation to go through the options and build it. If we put this in another function, then it will be more neat.